Mathlib/RingTheory/Derivation/MapCoeffs.lean

```diff
@@ -0,0 +1,98 @@
+/-
+Copyright (c) 2024 Daniel Weber. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Daniel Weber
+-/
+import Mathlib.RingTheory.Derivation.Basic
+import Mathlib.Algebra.Polynomial.Module.Basic
+
+/-!
+# Derivations of univariate polynomials
+
+In this file we prove properties about `R`-derivations of `Polynomial A`.
+-/
+
+noncomputable section
+
+open Polynomial Module
+
+namespace Derivation
+
+variable {R A M : Type*} [CommRing R] [CommRing A] [Algebra R A] [AddCommGroup M]
+  [Module A M] [Module R M] [IsScalarTower R A M] (d : Derivation R A M) (a : A)
+
+/--
+The `R`-derivation from `A[X]` to `M[X]` which applies the derivative to each
+of the coefficients.
+-/
+def mapCoeffs : Derivation R A[X] (PolynomialModule A M) where
```

Comment:
This name is ambiguous in that one could also derive the a multivariate polynomial coefficient-wise. What about something like this?
```suggestion
def polynomialCoeffs : Derivation R A[X] (PolynomialModule A M) where
```

Reply:
Should I also change the file name, or is it ok?

Reply:
Thinking about it a bit more, I don't love `polynomialCoeffs` - it feels weird to write `D.polynomialCoeffs p`

Reply:
I would personally just call it `Derivation.Polynomial`, but I don't know 🤷

Reply:
Perhaps just `Derivation.poly`? Or is that too confusing?

Reply:
In absolute, it might not be confusing. But here it would be confusing to the extent that existing files about `Polynomial` have `Polynomial` in their name, not just `Poly`.

Reply:
Perhaps `Polynomial.differentiateCoeffs`? Also that's still quite long

Reply:
This was originally in the same file as that derivation (`Algebra/Polynomial/Derivation`), but I moved it to this file

Reply:
Oh I see. I guess it can go back there

Reply:
@semorrison @erdOne could we get your opinions on this?

Reply:
In #14967 I also import `RingTheory.Derivation.DifferentialRing` to this file, so I think it's good to keep as a separate file

Reply:
How bad do you think this name is? I'm unable to think of any alternative I like

Reply:
I guess it's fine, but let's not resolve this conversation in case someone thinks of a better name